packages/terra-popup/src/_PopupUtils.js

```diff
@@ -10,124 +19,270 @@ const MIRROR_TB = {
 
-const switchAttachmentToRTL = (attachment) => {
-  const parsedValue = parseStringPair(attachment);
-  return `${parsedValue.vertical} ${MIRROR_LR[parsedValue.horizontal]}`;
+/**
+ * This function splits a given attachment into an object representing the corresponding vertical and horizontal values.
+ *
+ * @ param {string} attachment - The vertical-horizontal attachment value to split.
+ */
+const parseAttachment = (attachment) => {
+  if (!attachment) {
+    return { vertical: '', horizontal: '' };
+  }
+  const [vertical, horizontal] = attachment.split(' ');
+  return { vertical, horizontal: HORIZONTAL_HOOKSHOT_MAP[horizontal] };
 };
 
+/**
+ * This function returns an attachment object containing the mirrored attachment values.
+ *
+ * @ param {Object} attachment - The vertical and horizonal hookshot attachments.
+ */
 const mirrorAttachment = (attachment) => {
-  const parsedValue = parseStringPair(attachment);
-  let horizontal = parsedValue.horizontal;
-  let vertical = parsedValue.vertical;
-
-  if (isVerticalAttachment(parsedValue)) {
-    vertical = MIRROR_TB[parsedValue.vertical];
-  } else {
-    horizontal = MIRROR_LR[parsedValue.horizontal];
+  if (attachment.vertical !== 'middle') {
+    return { vertical: MIRROR_TB[attachment.vertical], horizontal: attachment.horizontal };
   }
-
-  return `${vertical} ${horizontal}`;
+  return { vertical: attachment.vertical, horizontal: MIRROR_LR[attachment.horizontal] };
 };
 
 /**
- * This method calculates a positional offset to be applied if the target is smaller than the arrow.
+ * This method calculates the horizontal offset to be applied to the content. Considers if the target's
+ * horizontal attachment is different than the contents attachment or the target's width is smaller than
+ * the arrow's position.
+ *
+ * @ param {Object} cAttachment - The vertical and horizonal hookshot attachments of the content.
+ * @ param {Object} tAttachment - The vertical and horizonal hookshot attachments of the target.
+ * @ param {func} targetNode - The target element for the popup to anchor to.
+ * @ param {number} arrowOffset - Half the base of the arrow.
+ * @ param {number} cornerOffset - The rounded corner size of the content.
  */
-const getContentOffset = (attachment, targetNode, arrowOffset, cornerOffset) => {
+const getContentOffset = (cAttachment, tAttachment, targetNode, arrowOffset, cornerOffset) => {
   const offset = { vertical: 0, horizontal: 0 };
+
   if (targetNode) {
-    if (isVerticalAttachment(attachment) && targetNode.clientWidth <= (arrowOffset * 2) + cornerOffset) {
-      if (attachment.horizontal === 'left') {
-        offset.horizontal = (arrowOffset + cornerOffset) - (targetNode.clientWidth / 2);
-      } else if (attachment.horizontal === 'right') {
-        offset.horizontal = -((arrowOffset + cornerOffset) - (targetNode.clientWidth / 2));
+    const segment = arrowOffset + cornerOffset;
+    if (isVerticalAttachment(cAttachment)) {
```

Comment:
Can we also use a guard clause here? It's clearer on what's happening without more nesting.
```
if (isVerticalAttachment(cAttachment)) == false) {
  return offset;
}
if (cAttachment.horizontal ! == tAttachment.horizontal) {
....
return offset;
```